src/custom-elements/src/components/button/button.tsx

```diff
@@ -0,0 +1,108 @@
+import { Component, Event, EventEmitter, Prop, Watch } from '@stencil/core'
+
+@Component({
+  tag: 'chi-button',
+  styleUrl: 'button.scss',
+  scoped: true
+})
+export class Button {
+  /**
+   *  to set button type { float, pill, close, icon }.
+   */
+  @Prop({ reflectToAttr: true }) type: string;
+
+  /**
+   *  to set button color { primary, secondary, danger, dark }.
+   */
+  @Prop({ reflectToAttr: true }) color: string;
+
+  /**
+   *  to set button size { sm, md, lg, xl }.
+   */
+  @Prop({ reflectToAttr: true }) size = 'md';
+
+  /**
+   *  to remove a buttons solid background and keep its colored border.
+   */
+  @Prop({ reflectToAttr: true }) outline = false;
```

Comment:
Add :string definition. In other PR we have changed the TSLint file to this rule.

Reply:
> @dani-cl-madrid I have made the necessary changes If you can review it then I'll start writing test for this and update the PR, TIA.

@srinath-koneru , please, first of all change the PR name in the way explained [here](https://github.com/CenturyLinkCloud/ux-chi/pull/279#issuecomment-494286357). Then, review the rest of comments. TY.